packages/react/src/primitives/SearchField/useSearchField.tsx

```diff
@@ -0,0 +1,75 @@
+import * as React from 'react';
+
+import { isFunction } from '../shared/utils';
+import { SearchFieldProps } from '../types';
+import { useComposeRefsCallback } from '../../hooks/useComposeRefsCallback';
+
+const ESCAPE_KEY = 'Escape';
+const ENTER_KEY = 'Enter';
+const DEFAULT_KEYS = [ESCAPE_KEY, ENTER_KEY];
+
+export const useSearchField = ({
+  onSubmit,
+  onClear,
+  externalRef,
+}: Partial<
+  SearchFieldProps & { externalRef: React.ForwardedRef<HTMLInputElement> }
```

Comment:
nitpick: can you create a separate type for hook props? (e.g `UseSearchFieldProps`)

```ts
export type UseSearchFieldProps = Partial<SearchFieldProps> & {
  externalRef?: React.ForwardedRef<HTMLInputElement>
};
```

Reply:
Fixed